lib/optimizely.rb

```diff
@@ -60,10 +62,14 @@ def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = n
         @logger.log(Logger::ERROR, e.message)
       end
 
-      @config_manager = StaticProjectConfigManager.new(datafile, @logger, @error_handler, skip_json_validation)
+      @config_manager = if config_manager.respond_to?(:get_config)
```

Comment:
If `sdk_key` is passed in, then we create a `HTTPProjectConfigManager` with it. Note, we should also be able to create a `DatafileProjectConfig` with the hard-coded datafile and pass it to the `HTTPProjectConfigManager` to have as its initial instance